### ISSUE TITLE
Registration

### ISSUE THREAD

**Issue by mikebattye** [asimshah/booking-app#45]
Failed to register a new member today at around 11:30

Call to "/user/register" failed: Internal Server Error

**Comment by asimshah** [comment]
Can you explain how to reproduce this error?

**Comment by mikebattye** [comment]
I assumed it was a problem with the membership lookup. I took a member's name and number from the existing site (one not already a member) and tried to register them.

**Comment by mikebattye** [comment]
Just tried this and there is definitely a problem with registration. I took some names and valid membership numbers. At least 3 gave me "No record found at the BMC" (eg carmelgray@yahoo.com G219646)

This one gave the internal server error: chris.rowlands@mac.com K340514

You should be able to try these yourself - as far as I am aware they are valid membership numbers.

**Comment by mikebattye** [comment]
I should add to this that it does correctly identify 'email already in use' and prevents you from registering again. I was able to reset the password for the email.

It might be useful to add after 'email in use' that there is an option to reset your password (though it is pretty obvious!)

**Comment by asimshah** [comment]
I have tried these two names and BMC numbers on my development system and they register ok. Looking at the test site in Weston, there appear to be multiple, perhaps serious, problems!

I need to know if this problem on the test site is 'permanent', i.e. no registrations work at all - or is it that some do and some don't. In either case, has this been this case since November 2016 (when this issue was recorded) or have successful registrations occurred in the mean time?

There is also some information in the log regarding Carmel Gray - the log records that the name 'Grey' was used which will fail as it is the wrong name.  I presume this is not the case with the other names but I can't check the logs for these as you have not given me any details.

**Comment by asimshah** [comment]
Further, I do not understand the comment about 'email already in use'. If the registration(s) have failed then the email addresses should not 'already be in use', surely?

**Comment by mikebattye** [comment]
Ignore the comment. It just means that the system is correctly picking up that the email is already assigned to a member. This of course is nothing to do with the lookup so I thought it might help to isolate the problem.

Regards, Mike

Sent from my iPad

On 21 Mar 2017, at 19:45, asimshah <notifications@github.com<mailto:notifications@github.com>> wrote:


Further, I do not understand the comment about 'email already in use'. If the registration(s) have failed then the email addresses should not 'already be in use', surely?

**Comment by mikebattye** [comment]
I haven't been able to register a new name on the current test site since the issue was first reported. I discovered it when I asked some real members to register just in case I was using incorrect numbers (or as you say missed typed names).

Regards, Mike

Sent from my iPad

On 21 Mar 2017, at 19:42, asimshah <notifications@github.com<mailto:notifications@github.com>> wrote:


I have tried these two names and BMC numbers on my development system and they register ok. Looking at the test site in Weston, there appear to be multiple, perhaps serious, problems!

I need to know if this problem on the test site is 'permanent', i.e. no registrations work at all - or is it that some do and some don't. In either case, has this been this case since November 2016 (when this issue was recorded) or have successful registrations occurred in the mean time?

There is also some information in the log regarding Carmel Gray - the log records that the name 'Grey' was used which will fail as it is the wrong name. I presume this is not the case with the other names but I can't check the logs for these as you have not given me any details.
